script.js

```diff
@@ -1,12 +1,125 @@
 //You can edit ALL of the code here
+const allEpisodes = getAllEpisodes().map(episode =>{
+  const formattedSeason = (episode.season.toString()).padStart(2, "0") 
+  const formattedNumber = (episode.number.toString()).padStart(2, "0")
+  const episodeCorrectFormatV2 = `S${formattedSeason}E${formattedNumber}`
+
+  return {
+    ...episode,
+    formattedNumber: episodeCorrectFormatV2
+  }
+});
+const rootEl = document.getElementById("container")
+const selectEl = document.getElementById("selectEl")
+const headerEl = document.getElementById("headerEl")
+const searchBar = document.getElementById("searchBar")
+const selector = document.getElementById("selector")
+const counter = document.getElementById("episodeCounter")
+
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  // makePageForEpisodes(allEpisodes);
+  createEpisodeCards(allEpisodes)
+  optionCreator(allEpisodes)
+  //restarting value of search input
+  searchBar.value = ""
+  //restarting Total episode text
+  counter.innerHTML = `Total episodes: ${allEpisodes.length}`
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+// episode option creator
+function optionCreator (listOfEpisodes) {
+  listOfEpisodes.forEach(episode => { 
+    const selectionOption = document.createElement("option")
+    selectionOption.innerHTML= `${episode.formattedNumber} - ${episode.name}`
+    selector.appendChild(selectionOption)
+  })
+  //add event listener to selector
+  selector.addEventListener("change",filterEpisodeBySelect)
+  function filterEpisodeBySelect() {
+    const filteredEpisodes = allEpisodes.filter(episode => {
+      return episode.name.includes(selector.value.substring(9))
+    }); createEpisodeCards(filteredEpisodes)
+    //change number of displaying episode
+    counter.innerHTML = `Displaying ${filteredEpisodes.length}`
```

Comment:
UX might benefit from consistency of what this should display e.g. always 'Displaying X/X episodes' (see that this is different on line 44 vs line 111 in your script for example.